### ISSUE TITLE
fail2ban regex how to create below mentioned haproxy log

### ISSUE THREAD

**Issue by juniorruby** [fail2ban/fail2ban#987]
Hi team,
              Have installed fail2ban in centos 6.5
i just want block the one who's trying to access this page by public ip address /myadmin/scripts/setup.php
can you pls guide how to 

Mar  7 07:26:16 localhost haproxy[9378]: 114.112.231.100:7594 [07/Mar/2015:07:26:13.316] http-ingress testexa/prodapp01 3046/0/1/16/3317 302 689 - - --VN 3/3/0/0/0 0/0 "GET /myadmin/scripts/setup.php HTTP/1.1"
Mar  9 02:12:59 localhost haproxy[9378]: 114.112.231.100:26518 [09/Mar/2015:02:12:55.901] http-ingress testexa/prodapp01 2999/0/0/19/3291 302 689 - - --VN 4/4/0/0/0 0/0 "GET /myadmin/scripts/setup.php HTTP/1.1"
Mar  9 02:13:06 localhost haproxy[9378]: 114.112.231.100:31767 [09/Mar/2015:02:13:06.157] http-ingress testexa/prodapp01 0/0/1/24/291 302 689 - - --VN 2/2/0/0/0 0/0 "GET /myadmin/scripts/setup.php HTTP/1.1

**Comment by Shaps** [comment]
Maybe something like this:
```
<HOST>:.+\"GET \/myadmin\/scripts\/setup\.php.+\"$
```

**Comment by sebres** [comment]
1) If you'll use it together with default filter/jail [haproxy-http-auth](https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/haproxy-http-auth.conf) just create `filter.d/haproxy-http-auth.local`
and add there:
```
[Definition]
blockurl = \/?(?:myadmin/scripts/setup\.php|another/evil/folder/)\S*
blockcode = (?:302|301)

failregex = %(known/failregex)s
            ^%(__prefix_line)s<HOST>:\S+ \[[^\]]+\] \S+ \S+ \d+/\d+/\d+/\d+/\d* %(blockcode)s [^"]* "[A-Z]+ %(blockurl)s HTTP/[^"]+"$
enabled = true
```

2) If not wanted or missing `haproxy-http-auth` (e. g. older version), just use following in your corresponding jail in `jail.local`:
```
[my-haproxy-jail]
blockurl = \/?(?:myadmin/scripts/setup\.php|another/evil/folder/)\S*
blockcode = (?:302|301)
# long universal version:
#failregex = ^(?:\[\])?\s*(?:<[^.]+\.[^.]+>\s+)?(?:\S+\s+)?(?:kernel: \[ *\d+\.\d+\]\s+)?(?:@vserver_\S+\s+)?(?:(?:(?:\[\d+\])?:\s+[\[\(]?haproxy(?:\(\S+\))?[\]\)]?:?|[\[\(]?haproxy(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)\s+)?(?:\[ID \d+ \S+\]\s+)?<HOST>:\S+ \[[^\]]+\] \S+ \S+ \d+/\d+/\d+/\d+/\d* %(blockcode)s [^"]* "[A-Z]+ %(blockurl)s HTTP/[^"]+"$
# short version:
failregex = ^\s*(?:\S+\s+)?(?:haproxy\[[^\]]+])?:?\s*<HOST>:\S+ \[[^\]]+\] \S+ \S+ \d+/\d+/\d+/\d+/\d* %(blockcode)s [^"]* "[A-Z]+ %(blockurl)s HTTP/[^"]+"$
...
```

In both cases you can control the list of block urls and block http codes using blockurl and blockcode sub-regex's (separated with pipe-char `|`).
